block_util/src/vhd.rs

```diff
@@ -0,0 +1,115 @@
+        // where the footer starts.
+        file.seek(SeekFrom::End(-512))?;
+
+        // Fill in the VhdFooter structure
+        let mut footer = [0; 512];
+        file.read_exact(&mut footer)?;
+
+        Ok(VhdFooter {
+            cookie: u64::from_be_bytes(footer[0..8].try_into().unwrap()),
+            features: u32::from_be_bytes(footer[8..12].try_into().unwrap()),
+            file_format_version: u32::from_be_bytes(footer[12..16].try_into().unwrap()),
+            data_offset: u64::from_be_bytes(footer[16..24].try_into().unwrap()),
+            time_stamp: u32::from_be_bytes(footer[24..28].try_into().unwrap()),
+            creator_application: u32::from_be_bytes(footer[28..32].try_into().unwrap()),
+            creator_version: u32::from_be_bytes(footer[32..36].try_into().unwrap()),
+            creator_host_os: u32::from_be_bytes(footer[36..40].try_into().unwrap()),
+            original_size: u64::from_be_bytes(footer[40..48].try_into().unwrap()),
+            current_size: u64::from_be_bytes(footer[48..56].try_into().unwrap()),
+            disk_geometry: u32::from_be_bytes(footer[56..60].try_into().unwrap()),
+            disk_type: u32::from_be_bytes(footer[60..64].try_into().unwrap()),
+            checksum: u32::from_be_bytes(footer[64..68].try_into().unwrap()),
+            unique_id: u128::from_be_bytes(footer[68..84].try_into().unwrap()),
+            saved_state: u8::from_be_bytes(footer[84..85].try_into().unwrap()),
+        })
+    }
+
+    pub fn cookie(&self) -> u64 {
+        self.cookie
+    }
+    pub fn features(&self) -> u32 {
+        self.features
+    }
+    pub fn file_format_version(&self) -> u32 {
+        self.file_format_version
+    }
+    pub fn data_offset(&self) -> u64 {
+        self.data_offset
+    }
+    pub fn time_stamp(&self) -> u32 {
+        self.time_stamp
+    }
+    pub fn creator_application(&self) -> u32 {
+        self.creator_application
+    }
+    pub fn creator_version(&self) -> u32 {
+        self.creator_version
+    }
+    pub fn creator_host_os(&self) -> u32 {
+        self.creator_host_os
+    }
+    pub fn original_size(&self) -> u64 {
+        self.original_size
+    }
+    pub fn current_size(&self) -> u64 {
+        self.current_size
+    }
+    pub fn disk_geometry(&self) -> u32 {
+        self.disk_geometry
+    }
+    pub fn disk_type(&self) -> u32 {
+        self.disk_type
+    }
+    pub fn checksum(&self) -> u32 {
+        self.checksum
+    }
+    pub fn unique_id(&self) -> u128 {
+        self.unique_id
+    }
+    pub fn saved_state(&self) -> u8 {
+        self.saved_state
+    }
+}
+
+/// Determine image type through file parsing.
+pub fn is_fixed_vhd(f: &mut File) -> bool {
+    let footer = match VhdFooter::new(f) {
+        Ok(footer) => footer,
+        Err(e) => {
+            warn!("VhdFooter couldn't be retrieved: {}", e);
+            return false;
+        }
+    };
+
+    footer.disk_type() == 0x2
+}
```

Comment:
This is perfect for unit testing. Have a reference footer sector and check the values that come out.

Reply:
Done!